### ISSUE TITLE
R2: State of field

### ISSUE THREAD

**Issue by cliffckerr** [sciris/sciris#442]
State-of-field comparison

I would like to see a comparison between odict and pandas.Series. They look very similar to me (ordered, key (aka) index based lookup, positional based lookup, index based slicing, and (implicit in odict) homogeneous types)
How does the datetime handling in sciris compare to the datetime support from pandas?
I would like to see a comparison with joblib and dask for sc_parallel
maybe also tqdm and rich for the progress bar and formatting tools?

**Comment by cliffckerr** [comment]
cc https://github.com/openjournals/joss-reviews/issues/5076

**Comment by cliffckerr** [comment]
To be honest, I struggled with this one: since the purpose of Sciris is to be broad rather than deep, there are many dozens of alternative libraries for different components of it. I did add mentions of Dask, Ray, and Celery, since all of these support cross-machine parallelization which Sciris (out of the box, via multiprocessing) does not; it looks like Joblib doesn't either. I've also added an example to the docstring of using Sciris *with* Dask. (Doesn't this just add an unnecessary intermediate step? Potentially, but `sc.parallelize()` does give a lot more flexibility with how arguments are passed than Dask does.)

While it felt like too much detail for the paper, I'm working on a PR that adds more detail to the docstrings of individual functions, including `sc.progressbar()` (rewriting to use tqdm, which had previously also been used in `sc.parallel_progress()`), `sc.colorize()` (mentioning rich), and `sc.date()` (comparing with pandas).

Re: `sc.odict` vs. `pd.Series`, while it's true that there are some similarities, `sc.odict` is intended to be a drop-in replacement for `dict`, which `pd.Series` is not (e.g., it doesn't have `keys`, `values`, or `items` methods). 

Re: dates, while pandas does have nice datetime support, the idea here is to make it easier to work with and interconvert dates of any format -- strings, Python datetime, etc. However, it's true that it could be easier to interconvert between date type, (including pandas -- issue [here](https://github.com/sciris/sciris/issues/464).
